core/lerna/commands/add-caching/index.js

```diff
@@ -0,0 +1,157 @@
+
+const { Command } = require("@lerna/command");
+const { writeJsonFile, readJsonFile, workspaceRoot, joinPathFragments } = require("@nrwl/devkit");
+const inquirer = require("inquirer");
+const log = require("npmlog");
+
+module.exports = factory;
+
+function factory(argv) {
+  return new AddCachingCommand(argv);
+}
+
+class AddCachingCommand extends Command {
+  constructor(argv) {
+    super(argv, { skipValidations: true });
+  }
+
+  initialize() {
+    if (this.options.useNx === false) {
+      this.logger.error(
+        "add-caching",
+        "The `add-caching` command is only available when using the Nx task runner"
+      );
+      // eslint-disable-next-line no-process-exit
+      process.exit(1);
+    }
+
+    const packages = this.packageGraph?.rawPackageList || [];
+    const uniqueScriptNames = new Set();
+    for (const pkg of packages) {
+      for (const scriptName of Object.keys(pkg.scripts || {})) {
+        uniqueScriptNames.add(scriptName);
+      }
+    }
+    this.uniqueScriptNames = Array.from(uniqueScriptNames);
+  }
+
+  async execute() {
+    this.logger.info(
+      "add-caching",
+      "Please answer the following questions about the scripts found in your workspace in order to generate task runner configuration"
+    );
+    process.stdout.write("\n");
+
+    log.pause();
+
+    const { targetDefaults } = await inquirer.prompt([
+      {
+        type: "checkbox",
+        name: "targetDefaults",
+        message: "Which of the following scripts need to be run in deterministic/topoglogical order?\n",
+        choices: this.uniqueScriptNames,
+      },
+    ]);
+
+    const { cacheableOperations } = await inquirer.prompt([
+      {
+        type: "checkbox",
+        name: "cacheableOperations",
+        message:
+          "Which of the following scripts are cacheable? (Produce the same output given the same input)\n",
+        choices: this.uniqueScriptNames,
+      },
+    ]);
+
+    const scriptOutputs = {};
+
+    for (const scriptName of cacheableOperations) {
+      // eslint-disable-next-line no-await-in-loop
+      scriptOutputs[scriptName] = await inquirer.prompt([
+        {
+          type: "input",
+          name: scriptName,
+          message: `Where within a particular project directory does the "${scriptName}" script write its outputs?\n`,
+          default: "dist",
+          validate: (input) => {
+            if (!input) {
+              return "Please provide a project relative path";
+            }
+            return true;
+          },
+        },
+      ]);
+    }
+
+    log.resume();
+
+    process.stdout.write("\n");
+
+    this.convertAnswersToNxConfig({ cacheableOperations, targetDefaults, scriptOutputs });
+
+    this.logger.success("add-caching", "Successfully updated task runner configuration in `nx.json`");
+
+    this.logger.info(
+      "add-caching",
+      "Learn more about task runner configuration here: https://lerna.js.org/docs/concepts/task-pipeline-configuration"
```

Comment:
I'd also add a note saying --sort, --no-sort and --parallel no longer work and link to the doc we link from the warning.